### PR TITLE
UI: Adds rating

### PR DESCRIPTION
**Pull request by pratap0007**
This patch adds:
    - MST store for rating
    - User can rate a particular resource
    - User can see their previous rating
    - User can update their rating
    - To rate a resource user needs to be authenticated to Hub

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/Rating/index.tsx [hunk 1]
```diff
@@ -1,24 +1,108 @@
...
+  if (!user.isLoading) {
+    highlightStar(user.userRating);
+    user.setLoading(true);
+  }
+
+  const starList = (
+    <ul className="hub-rate-area">
+      <input readOnly type="radio" id="5" name="rating" value="5" checked={star[4]} />
+      <label htmlFor="5">
+        <StarIcon />
```
Comment (vinamra28) [inline ui/src/containers/Rating/index.tsx:49]: can we use `StartIcon` from Icons component?

File: ui/src/containers/Rating/index.tsx [hunk 2]
```diff
@@ -1,24 +1,108 @@
...
-const Rating: React.FC = () => {
+const Rating: React.FC = observer(() => {
+  const { user, resources } = useMst();
+  const { name } = useParams();
+  const history = useHistory();
+
+  const [star, setStar] = useState([false, false, false, false, false]);
+
+  const highlightStar = (value: number) => {
+    const x = [false, false, false, false, false];
```
Comment (vinamra28) [inline ui/src/containers/Rating/index.tsx:18]: can we use a better variable over here instead of `x`?

File: ui/src/containers/Rating/Rating.test.tsx [hunk 3]
```diff
@@ -1,8 +1,110 @@
...
+    expect(component.debug()).toMatchSnapshot();
+  });
+
+  it('should find star Icon', () => {
+    const component = mount(
+      <Provider>
+        <Rating />
+      </Provider>
+    );
+    expect(component.find('input').length).toBe(5);
```
Comment (vinamra28) [inline ui/src/containers/Rating/Rating.test.tsx:43]: instead of checking for `input` can we check for the `StarIcon` ?

File: ui/src/store/auth.test.ts [hunk 4]
```diff
@@ -60,4 +60,58 @@ describe('Store functions', () => {
       }
     );
   });
+
+  it('can get the rating for a resource', (done) => {
```
Comment (PuneetPunamiya) [inline ui/src/store/auth.test.ts:64]: Can we change it to get `user's rating for a resource` instead of rating for a resource

File: ui/src/store/auth.test.ts [hunk 5]
```diff
@@ -60,4 +60,58 @@ describe('Store functions', () => {
       }
     );
   });
+
+  it('user\'s rating for buildah resource', (done) => {
```
Comment (PuneetPunamiya) [inline ui/src/store/auth.test.ts:64]: nit: Can we please have a proper heading please

File: ui/public/config.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
 window.config = {
-  API_URL: 'https://api.hub.tekton.dev',
-  GH_CLIENT_ID: ''
-};
+  API_URL: 'http://localhost:8000',
+  GH_CLIENT_ID: "60685581b4890c56fabd"
```
Comment (PuneetPunamiya) [inline ui/public/config.js:3]: Should we keep the client id as empty string now ???

File: ui/public/config.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
 window.config = {
-  API_URL: 'https://api.hub.tekton.dev',
-  GH_CLIENT_ID: ''
-};
+  API_URL: 'http://localhost:8000',
```
Comment (PuneetPunamiya) [inline ui/public/config.js:2]: Can we revert this change by having the `hub-api` url

File: ui/src/containers/Rating/index.tsx [hunk 8]
```diff
@@ -1,24 +1,114 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { observer } from 'mobx-react';
+import { useHistory, useParams } from 'react-router-dom';
 import { StarIcon } from '@patternfly/react-icons';
+import { IconSize } from '@patternfly/react-icons';
```
Comment (PuneetPunamiya) [inline ui/src/containers/Rating/index.tsx:5]: > import { StarIcon, IconSize } from '@patternfly/react-icons';

File: ui/src/containers/Rating/index.tsx [hunk 9]
```diff
@@ -1,24 +1,114 @@
...
+import { Alert, AlertActionCloseButton, AlertGroup } from '@patternfly/react-core';
+import { useMst } from '../../store/root';
+import { assert } from '../../store/utils';
 import './Rating.css';
 
-const Rating: React.FC = () => {
+const Rating: React.FC = observer(() => {
+  const { user, resources } = useMst();
```
Comment (PuneetPunamiya) [inline ui/src/containers/Rating/index.tsx:12]: Can we move this down on line no 41
and `const {resources} = useMst()` on line 71

File: ui/src/containers/Rating/index.tsx [hunk 10]
```diff
@@ -1,24 +1,114 @@
...
+  const { name } = useParams();
+  const history = useHistory();
```
Comment (PuneetPunamiya) [inline ui/src/containers/Rating/index.tsx:14]: Also can you move history down on before rate resource function

**Review by PuneetPunamiya — COMMENTED** [review]
Thank you for the patch, the pr looks fine to me only some minor changes to address :call_me_hand:

File: ui/src/containers/Rating/index.tsx [hunk 11]
```diff
@@ -1,24 +1,114 @@
...
+  if (!user.isLoading) {
+    highlightStar(user.userRating);
+    user.setLoading(true);
+  }
+
+  const starList = (
+    <ul className="hub-rate-area">
+      <input readOnly type="radio" id="5" name="rating" value="5" checked={star[4]} />
+      <label htmlFor="5">
+        <StarIcon id="5" size={IconSize.sm} label="5" />
```
Comment (vinamra28) [inline ui/src/containers/Rating/index.tsx:51]: can we do it like  `<Icon id={Icons.Star} size={IconSize.sm} label="5" />` ?

File: ui/src/containers/BasicDetails/BasicDetails.css [hunk 12]
```diff
@@ -89,3 +89,6 @@
   position: relative;
   left: 3em;
 }
+.hub-details-average-rating svg{
+  color: #484848;
+}
```
Comment (vinamra28) [inline ui/src/containers/BasicDetails/BasicDetails.css:94]: can we add one extra line below

File: ui/src/store/testdata/rating_resource_13.json [hunk 13]
```diff
@@ -0,0 +1 @@
+{ "rating": 2 }
```
Comment (PuneetPunamiya) [inline ui/src/store/testdata/rating_resource_13.json:1]: Since it is a single line data should we add a new file for this or can we just use a constant ?? :thinking:  wdyt ?

File: ui/src/containers/UserProfile/index.tsx [hunk 14]
```diff
@@ -18,7 +18,8 @@ const UserProfile: React.FC = () => {
   const logout = () => {
     localStorage.clear();
```
Comment (PuneetPunamiya) [inline ui/src/containers/UserProfile/index.tsx:19]: Can we please move this all to store by creating an action ?
